README.md

```diff
@@ -4,22 +4,287 @@
 [![Code Climate](https://codeclimate.com/github/Dhii/module-interface/badges/gpa.svg)](https://codeclimate.com/github/Dhii/module-interface)
 [![Test Coverage](https://codeclimate.com/github/Dhii/module-interface/badges/coverage.svg)](https://codeclimate.com/github/Dhii/module-interface/coverage)
 [![Latest Stable Version](https://poser.pugx.org/dhii/module-interface/version)](https://packagist.org/packages/dhii/module-interface)
-[![This package complies with Dhii standards](https://img.shields.io/badge/Dhii-Compliant-green.svg?style=flat-square)][Dhii]
 
 ## Details
 This package contains interfaces that are useful in describing modules and their attributes and behaviour.
-
-### Interfaces
-- [`ModuleInterface`][] - The interface for a module. A module is an object that represents an
-application fragment. Modules are prepared using `setup()`, which returns a `ServiceProviderInterface` instance that
-the application may consume, and invoked using `run()`.
  
 ### Requirements
 - PHP: 7.1 and up, until 8.
 
     Officially supports at least up to php 7.4.x.
 
+### Interfaces
+- [`ModuleInterface`][] - The interface for a module. A module is an object that represents an
+application fragment. Modules are prepared using `setup()`, which returns a `ServiceProviderInterface` instance that
+the application may consume, and invoked using `run()`, consuming the application's DI container.
+- [`ModuleAwareInterface`][] - Something that can have a module retrieved.
+- [`ModuleExceptionInterface`][] - An exception thrown by a module.
+
+### Usage
+#### Module Package
+In your module's pacakge, create a file that returns a module factory. This factory MUST return an instance
+of `ModuleInterface` from this pacakge. By convention, this file has
+the name `module.php`, and is located in the root directory. Below is a very basic example. In real life,
+the service provider and the module will often have named classes of their own, and factories and extensions
+will be located in `services.php` and `extensions.php` respectively, by convention.
```

Comment:
Module authors can still get as creative as they want, because nobody is forcing them to follow the convention. These files can really be anywhere and be called anything, because only the module itself interacts with them. However, it makes it faster for other module users to understand the module.

A special case is `module.php`. It contains the module factory function. This convention is useful if an application wants to specify modules by name such as package name. For this to work, all module files have to have the same name and relative location.

I was recently asked by several people to describe how to use the module system, and that's what I did here, because people are confused about how to get started with it. I agree that this all convention and opinion. Ultimately, it belongs in some technical post written by one of us and shared in relevant dev channels. But for now, this is the centralized location which everyone will look at, and they will know how to gets started quickly.